runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -426,7 +426,7 @@ bool J9::ValuePropagation::transformIndexOfKnownString(
          else
             {
             uintptr_t element = TR::Compiler->om.getAddressOfElement(comp(), string, start + TR::Compiler->om.contiguousArrayHeaderSizeInBytes());
-            int8_t chByte  = *((uint8_t*)element);
+            uint8_t chByte  = *((uint8_t*)element);
             ch = chByte;
```

Comment:
The same function does `ch = *((uint16_t*)element);` for the `is16Bit` case, and it does not extend the sign in the `uint16_t` -> `int32_t` conversion.
Maybe we can do in the same way for the 8-bit path, removing the intermediate variable `chByte`.

Reply:
Eliminated the intermediate variable as suggested.